src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Controllers/Api/Statistics/BoundaryLevelControllerTests.cs

```diff
@@ -0,0 +1,178 @@
+        await TestApp.AddTestData<StatisticsDbContext>(context =>
+        {
+            context.BoundaryLevel.Add(
+                new()
+                {
+                    Id = 1,
+                    Created = DateTime.Now,
+                    Label = "Boundary Level 1",
+                    Level = GeographicLevel.Country,
+                    Published = DateTime.Now,
+                });
+        });
+
+        var client = TestApp
+            .SetUser(AuthenticatedUser(RoleClaim(RoleNames.BauUser)))
+            .CreateClient();
+
+        var request = new BoundaryLevelUpdateRequest
+        {
+            Id = 1,
+            Label = "Updated Boundary Level 1",
+        };
+
+        var response = await client.PatchAsync("api/boundary-level", JsonContent.Create(request));
+
+        response.AssertNoContent();
+
+        await using var context = TestApp.GetDbContext<StatisticsDbContext>();
+
+        var saved = await context.BoundaryLevel.SingleAsync(bl => bl.Id == request.Id);
+
+        Assert.Multiple(
+            () => Assert.Equal(request.Id, saved.Id),
+            () => Assert.Equal(request.Label, saved.Label),
+            () => Assert.NotNull(saved.Updated)
+        );
+    }
+
+    [Fact]
+    public async Task CreateBoundaryLevel_Success()
+    {
+        await TestApp.AddTestData<StatisticsDbContext>(context =>
+        {
+            context.BoundaryLevel.Add(
+                new()
+                {
+                    Id = 1,
+                    Created = DateTime.Now,
+                    Label = "Boundary Level 1",
+                    Level = GeographicLevel.Region,
+                    Published = DateTime.Now,
+                });
+        });
+
+        var client = TestApp
+            .SetUser(AuthenticatedUser(RoleClaim(RoleNames.BauUser)))
+            .CreateClient();
+
+        var path = Path.Combine(Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location)!, @"REG - Regions England BUC 202212.geojson");
+        var content = new MultipartFormDataContent
+        {
+            { new StringContent("Region"), "level" },
+            { new StringContent("Boundary Data 1"), "label" },
+            { new StreamContent(File.OpenRead(path)), "file", "REG - Regions England BUC 202212.geojson" },
+            { new StringContent("2020-01-01"), "published" },
+        };
+
+        var response = await client.PostAsync("api/boundary-level", content);
+
+        response.AssertNoContent();
+
+        await using var context = TestApp.GetDbContext<StatisticsDbContext>();
+
+        var savedBoundaryLevel = await context.BoundaryLevel.SingleAsync(bl => bl.Id == 1);
+        var savedBoundaryData = await context.BoundaryData.Include(bd => bd.BoundaryLevel).FirstAsync();
+
+        Assert.Multiple(
+            () => Assert.NotNull(savedBoundaryLevel),
+            () => Assert.NotNull(savedBoundaryData),
+            () => Assert.Equal(GeographicLevel.Region, savedBoundaryLevel.Level)
+        );
```

Comment:
I'm not sure the `Include(bd => bd.BoundaryLevel)` is doing anything here? I think .Where(bd => db.BoundaryLevel.Id == 1)` would be more appropriate.

`FirstAsync` is going to blow up if there aren't any so I think that makes the check `Assert.NotNull(savedBoundaryData)`  redundant too. Maybe you could assert on the count `BoundaryData` for that level?